package/src/components/MessageInput/CommandsButton.tsx

```diff
@@ -1,73 +1,76 @@
-import React from 'react';
-import type { GestureResponderEvent } from 'react-native';
-import { Pressable } from 'react-native';
+import React, { useCallback, useMemo } from 'react';
+import type { GestureResponderEvent, PressableProps } from 'react-native';
+import { Pressable, View } from 'react-native';
 
-import {
-  isSuggestionCommand,
-  SuggestionsContextValue,
-  useSuggestionsContext,
-} from '../../contexts/suggestionsContext/SuggestionsContext';
+import { SearchSourceState, TextComposerState } from 'stream-chat';
+
+import { useMessageComposer } from '../../contexts/messageInputContext/hooks/useMessageComposer';
 import { useTheme } from '../../contexts/themeContext/ThemeContext';
+import { useStateStore } from '../../hooks/useStateStore';
 import { Lightning } from '../../icons/Lightning';
 
-type CommandsButtonPropsWithContext = Pick<SuggestionsContextValue, 'suggestions'> & {
+export type CommandsButtonProps = {
   /** Function that opens commands selector */
-  handleOnPress?: ((event: GestureResponderEvent) => void) & (() => void);
+  handleOnPress?: PressableProps['onPress'];
 };
 
-const CommandsButtonWithContext = (props: CommandsButtonPropsWithContext) => {
-  const { handleOnPress, suggestions } = props;
+const textComposerStateSelector = (state: TextComposerState) => ({
+  suggestions: state.suggestions,
+  text: state.text,
+});
+
+const searchSourceStateSelector = (nextValue: SearchSourceState) => ({
+  items: nextValue.items,
+});
+
+export const CommandsButton = (props: CommandsButtonProps) => {
+  const { handleOnPress } = props;
+  const messageComposer = useMessageComposer();
+  const { textComposer } = messageComposer;
+  const { suggestions, text } = useStateStore(textComposer.state, textComposerStateSelector);
```

Comment:
Relying on changes to `text` here means that every time the text changes, this thing will try to rerender. We only really need to know if there is `text` currently or not - we don't care about every text change. Same goes for all other places where `text` is used but its value is not relevant (and only having/not having `text` is)

Reply:
I think I can use `textComposer.isTextEmpty` in that case.

Reply:
I don't have a immediate fix for this but I will check.

Reply:
This one won't be an issue, in my opinion, because we immediately return `null` if there is no text. Should be fine as we are getting and listening to the text within the component(not coming through a prop or so).

Reply:
We do, yes - but the entire component will rerender every time text changes even though we are only really interested in whether it exists or not, that's what I mean. Can you please try it out and see ? The commands button shouldn't rerender this many times for nothing, but rather it should only appear/disappear based on what we have.